hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -145,16 +235,29 @@ public RocksDBCheckpointDiffer(String dbPath,
     //  mapping.
   }
 
+  /**
+   * Helper function that recursively deletes the dir. TODO: REMOVE
+   */
+  boolean deleteDirectory(File directoryToBeDeleted) {
+    File[] allContents = directoryToBeDeleted.listFiles();
+    if (allContents != null) {
+      for (File file : allContents) {
+        deleteDirectory(file);
+      }
+    }
+    return directoryToBeDeleted.delete();
+  }
+
   // Node in the DAG to represent an SST file
   private class CompactionNode {
-    public String fileName;   // Name of the SST file
-    public String snapshotId; // The last snapshot that was created before this
+    private String fileName;   // Name of the SST file
+    private String snapshotId; // The last snapshot that was created before this
     // node came into existance;
-    public long snapshotGeneration;
-    public long totalNumberOfKeys;
-    public long cumulativeKeysReverseTraversal;
+    private long snapshotGeneration;
+    private long totalNumberOfKeys;
+    private long cumulativeKeysReverseTraversal;
 
-    CompactionNode (String f, String sid, long numKeys, long compactionGen) {
+    CompactionNode(String f, String sid, long numKeys, long compactionGen) {
       fileName = f;
       snapshotId = sid;
       snapshotGeneration = lastSnapshotCounter;
```

Comment:
It seems to incorrect to me to have the snapshotGeneration be set by the lastSnapshotCounter at the time the node is created.  To me it should be the value of the lastSnapshotCounter at the time the compaction log file was created, (in other words it probably should be stored and read from the compaction log.)

Reply:
You are right. Should be fix by e76a24eddefcc44ab429f2be3245e0ec252c7d5a now.